lib/rubocop/cop/style/auto_resource_cleanup.rb

```diff
@@ -16,31 +16,38 @@ module Style
       #   File.open('file') do |f|
       #     # ...
       #   end
+      #
+      #   # bad
+      #   f = Tempfile.open('temp')
+      #
+      #   # good
+      #   Tempfile.open('temp') do |f|
+      #     # ...
+      #   end
       class AutoResourceCleanup < Base
-        MSG = 'Use the block version of `%<class>s.%<method>s`.'
-
-        TARGET_METHODS = { File: :open }.freeze
+        MSG = 'Use the block version of `%<current>s`.'
+        RESTRICT_ON_SEND = %i[open].freeze
 
-        RESTRICT_ON_SEND = TARGET_METHODS.values.freeze
+        # @!method file_open_method?(node)
+        def_node_matcher :file_open_method?, <<~PATTERN
+          (send (const nil? {:File :Tempfile}) :open ...)
```

Comment:
Should we also detect the fully qualified constants?

```suggestion
          (send (const {nil? cbase} {:File :Tempfile}) :open ...)
```